src/mo_def/syntax.ml

```diff
@@ -193,7 +193,7 @@ and exp' =
   | AnnotE of exp * typ                        (* type annotation *)
   | ImportE of (string * resolved_import ref)  (* import statement *)
   | ThrowE of exp                              (* throw exception *)
-  | TryE of exp * case list * exp option       (* catch exception / finally *)
+  | TryE of exp * case option * exp option     (* catch exception / finally *)
```

Comment:
Well, I am not so confident about whether this change is worth it. @crusso what do you think?